syft/pkg/cpe_test.go

```diff
@@ -105,6 +109,9 @@ func Test_CPEParser(t *testing.T) {
 			assert.Equal(t, c1, test.WFN)
 			assert.Equal(t, c2, test.WFN)
 			assert.Equal(t, CPEString(test.WFN), test.CPEString)
+			assert.Equal(t, CPEURI(c2), test.CPEUrl)
+			assert.Equal(t, CPEURI(c1), test.CPEUrl)
+			assert.Equal(t, must(NewCPE(CPEURI(c1))), test.WFN)
```

Comment:
This ensures that we are also making our URL generation also go through the same rigorous CPE encoding/decoding test that our CPE strings go through.